### ISSUE TITLE
Failed to decode message from payload

### ISSUE THREAD

**Issue by Eugene282** [Consensys-Incorporated/quorum#379]
#### System information

Geth version: `1.7.2`

OS & Version: Debian 9

Branch, Commit Hash or Release: `master d4a1b09 `

#### Actual behaviour

We use in our project -JP Morgan Quorum (Ethereum based) bloackchain - with istanbul voting mode. We have 4 nodes (3 of them are validators - hard coded, nonchangable)

And from time to time (unpredictable - but approximately once per several hours) - and the less nodes we use - the more rare the failure occurs. (for 7 nodes - once per two days). Also quorum node may consume up to 1Gb of memory after sending a Transaction to a contract, deployed on Quorum. (Transaction just sets two uint storage variables)

#### Questions

And I have two questions: 1. Did somebody have the same issue and how resolved it? 2. Did somebody use Quorum in production, or it is not stable enough for that purpose? 3. Have somebody guess or idea about such kind of failure? 4. Is it normal for Quorum to consume a lot of memory

I know, questions are "common and abstract". But I need, besides tha issue solution, to have your opinion and view on Quorum.

And our genesis.json file, if it helps (all the nodes listed there, we, of course, have configured and launched)

{
    "alloc": {
     "0xed9d02e382b34818e88b88a309c7fe71e65f419d": {
        "balance": "1000000000000000000000000000"
      },
      "0xca843569e3427144cead5e4d5999a3d0ccf92b8e": {
        "balance": "1000000000000000000000000000"
      },
      "0x0fbdc686b912d7722dc86510934589e0aaf3b55a": {
        "balance": "1000000000000000000000000000"
      },
      "0x9186eb3d20cbd1f5f992a950d808c4495153abd5": {
        "balance": "1000000000000000000000000000"
      },
      "0x0638e1574728b6d862dd5d3a3e0942c3be47d996": {
        "balance": "1000000000000000000000000000"
      }
    },
    "coinbase": "0x0000000000000000000000000000000000000000",
    "config": {
      "homesteadBlock": 1,
        "eip150Block": 2,
        "eip150Hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
        "eip155Block": 3,
        "eip158Block": 3,
      "istanbul": {
        "epoch": 30000,
        "policy": 0
      },
      "isQuorum": true
    },
    "extraData": "0xd783010702846765746887676f312e392e33856c696e75780000000000000000f8c9f83f948157d4437104e3b8df4451a85f7b2438ef6699ff94b912de287f9b047b$
    "gasLimit": "0xFFFFFFFFFFFFFFF0",
    "difficulty": "0x1",
    "mixHash": "0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365",
    "nonce": "0x0",
    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "timestamp": "0x00"
  }

#### Steps to reproduce the behaviour

Occasional. Hard reproduce it in a predicatble way

#### Backtrace

````
INFO [05-07|10:14:12] Successfully sealed new block lock number=254599 hash=a0f9f2…c10589
INFO [05-07|10:14:12] :hammer: mined potential block n number=254599 hash=a0f9f2…c10589
INFO [05-047|10:14:12] Commit new mining work deduped=254600 txs=0 uncles=0 elapsed=2.774ms
ERROR[05-07|10:18:42] Failed to decode message from payload address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 e
rr="unauthorized address"
````

**Comment by fixanoid** [comment]
Hi @Eugene282 here are some answers. Hopefully, @yutelin or @markya0616 could chime in about Istanbul errors as well if needed.

2. Did somebody use Quorum in production, or it is not stable enough for that purpose?
I'm aware of at least 3 projects using Quorum in prod, thing is tho, They're either using older QuorumChain consensus or newer Raft based one -- I'd be interested to hear about Istanbul in prod as well.

1. Did somebody have the same issue and how resolved it?
3. Have somebody guess or idea about such kind of failure?
For Istanbul, you have specific instructions to order validators correctly. Here is the repo for istanbul tools that's helpful to generate required genesis file: https://github.com/getamis/istanbul-tools. On top of this, when a new validator becomes available in the cluster, other nodes need to accept it as per this article: https://medium.com/getamis/istanbul-bft-ibft-c2758b7fe6ff -- see the istanbul.propose API. 

4. Is it normal for Quorum to consume a lot of memory
This depends on the use, but in essence, yes, Quorum/Geth are memory intensive apps.

Please update follow up if at all possible if you were able to solve the issue.

**Comment by fixanoid** [comment]
Ahh, forgot Alastria which is running Istanbul in prod: https://alastria.io

**Comment by Eugene282** [comment]
Thanks for reply. Very glad to learn, that Quorum is used in production) 

What about the issue - Yes, I know propose API and use it. But the issue occurs even when I just launch blockchain and do not touch anything else. It works for a while (several hours) and then fails. I've checked it on 4 nodes (with 3 validators), and 7 nodes (with 7 validators). 

Very strange that it works and then fails (not fails immediately - if I did something wrong with launching process). 

So, I'm interesting (may be your guess and thoughts) - what teoretically could produce "Failed to decode..." error message on the working nodes?

**Comment by renuseabhaya** [comment]
@Eugene282  I got the same error as "Failed to decode message from payload address=0xd8Dba507e85F116b1f7e231cA8525fC9008A6966 err="unauthorized address" , so are you using genesis file created by istanbul-tools or creating config.toml(which has accounts created by you) file to create genesis.json??

**Comment by Eugene282** [comment]
@renuseabhaya, we are getting that error even for quorum 7nodes example, which is published here, on jpmorgan's github.

**Comment by Eugene282** [comment]
Hello. We still have the same issue. Genesis is generated by getamis/istanbul-tools

Genesis file and logs of our 4 nodes are attached.
[genesis.json.txt](https://github.com/jpmorganchase/quorum/files/2275336/genesis.json.txt)
[bug-logs.tar.gz](https://github.com/jpmorganchase/quorum/files/2275422/bug-logs.tar.gz)

Blockchain stucks with "Failed to decode..." message. After restarting one or several nodes (not clear what exactly node should be restarted to wake up blockchain) blockchain continue generate blocks. After some time (minutes or hours) it stucks again.

Any help or advices, please.

**Comment by MariusVanDerWijden** [comment]
Is anyone working on this? I am having the same error.

**Comment by renuseabhaya** [comment]
@Eugene282 @MariusVanDerWijden 
Failed to decode..." message problem is solved..And able to create istanbul network & deployed  contracts both publickly & using privateFor. But blockNumber keeps on increasing ,actually it's expected behaviour

**Comment by Eugene282** [comment]
@renuseabhaya 
You mean this bug should not appear in latest version of Quorum?

And I didn't quit understand what does it mean "able to create network" and "but blockNumber keeps on increasing" - there was no problem (in context of this issue) with creating network itself, and of course block number increases in working blockchain. Or you mean, that is the result of bug fixing and blockchain does not (and will never) stuck with "Failed to decode " message? 
(sorry for misunderstanding)

**Comment by renuseabhaya** [comment]
@Eugene282  I used istanbul tools for configuring Istanbul network & from new eth account, I deplyed contracts . This is how my 'Failed to decode...' problem solved. 
And regarding  blocknumber increase, if there are no txns in queue, blocks will no be created but in Istanbul consensus mode blocks are created at 1sec intervals

**Comment by Eugene282** [comment]
@renuseabhaya Well, I am glad that you solved your issue, but what about mine one?) I used istanbul tools with new accounts as well - and "failed to decode..." happens from time to time. Looks like you propose just do some magic with using istanbul tools and hope that it solve the problem, but don't want to try understand the nature of this bug - for example - what it does even mean "Failed to decode..." and how it became possible?
I don't know what to do - I just have this bug, appearing occasionally, and even can't guess what is the reason - wrong node, wrong network configuration, wrong environment, or something else.
May be you have any guess?

**Comment by renuseabhaya** [comment]
@Eugene282 Can you please list down all the steps that you have tried..?

**Comment by lutianzhou001** [comment]
I have the same problem who can help me ....

**Comment by fixanoid** [comment]
Closing as the subset of these issues has been addressed in v2.1.x. Participants, please create a new ticket if the issue has been encountered again. Many thanks.
